src/jwtMiddleware.ts

```diff
@@ -88,53 +94,60 @@ export const jwtMiddleware = <
 >({
   key,
   algorithm,
-  critHandlers,
   customMessages = {},
   ignorePatterns,
-  onSuccess,
-  onFailure,
+  onSuccess = () => {},
+  onFailure = () => true,
 }: JwtMiddlewareOptions): T => {
   Object.assign(customMessages, errorMessages);
 
   const core: RouterMiddleware = async (ctx, next) => {
-    if (!ignorePatterns || !ignorePath(ctx, ignorePatterns)) {
-      let isUnauthorized = true;
-      let validatedJwt;
-      if (ctx.request.headers.has("Authorization")) {
-        const authHeader = ctx.request.headers.get("Authorization")!;
-
-        if (authHeader.startsWith("Bearer ") && authHeader.length > 7) {
-          const jwt = authHeader.slice(7);
-
-          validatedJwt = await validateJwt({
-            jwt,
-            key,
-            algorithm,
-            critHandlers,
-          });
-
-          if (validatedJwt.isValid) {
-            isUnauthorized = false;
-            onSuccess && onSuccess(ctx, validatedJwt);
-          }
-        }
+    const onUnauthorized = async (
+      jwtValidation: Error,
+      isJwtValidationError: boolean = false,
```

Comment:
I've called this `isJwtValidationError` because I didn't middleware users want it to be too much coupled with the underlying library (djwt). But I'm not sure I fully understood your intentions and thus I haven't changed much of the way errors were handled here

Reply:
so what I meant with this is that if this arg is true, the message on line 121, should be the actual error message rather than one of the custom messages, i.e. `isJwtValidationError ? jwtValidation.message :  customMessages?.ERROR_INVALID_AUTH,`